layers/meta-tegrademo/classes/fitimage-workarounds.bbclass

```diff
@@ -0,0 +1,16 @@
+# Cleaning in /boot folder.
+# Only FIT image should be installed in this directory.
+# the FIT image installed in the rootfs should have Linux kernel + DTB + initramfs
+
+do_fitimage_cleaning () {
+    bbnote "Remove all unneccesary files: Image, fitImage, dtb, initrd"
+    if [ -d "${IMAGE_ROOTFS}/boot" ]; then
+        rm -f ${IMAGE_ROOTFS}/boot/Image*
+        rm -f ${IMAGE_ROOTFS}/boot/fitImage*
+        rm -f ${IMAGE_ROOTFS}/boot/*.dtb
+        rm -f ${IMAGE_ROOTFS}/boot/initrd
+    fi
+
+}
+
+addtask fitimage_cleaning after do_rootfs before do_image_ext4
```

Comment:
Instead of a separate task, could you just append this to ROOTFS_POSTPROCESS_COMMAND? 

Even better would be to eliminate the need for the workaround, of course. :)

Reply:
It works fine when using `ROOTFS_POSTPROCESS_COMMAND`, I didn't succeed without adding the  workaround because of the `RDEPENDS_${KERNEL_PACKAGE_NAME}-base`
Please take a look here (Yocto project documentation): https://docs.yoctoproject.org/3.1.5/ref-manual/migration-2.5.html#miscellaneous-changes

Reply:
Did you try just modifying the `RDEPENDS_${KERNEL_PACKAGE_NAME}-base` setting in  the `linux-tegra_4.9.bb` recipe so it's null when using FIT images?

Reply:
I try it in the machine conf file. Is that wrong ?
I will try to update the bbappend or create it ?
Thanks a lot for your help :)

Reply:
Hmm. A bbappend sounds like the right place, we'd just need to figure out the right conditions under which to clear the dependency (assuming it works).

Reply:
Right

Reply:
I tried by adding the following line in the `linux-tegra` bbappend recipe but it doesn't work.

`RDEPENDS_${KERNEL_PACKAGE_NAME}-base = ""`

I will keep the usage of `ROOTFS_POSTPROCESS_COMMAND`